cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/prometheus/PrometheusMetricSampler.java

```diff
@@ -104,18 +107,16 @@ private void configurePrometheusAdapter(Map<String, ?> configs) {
             throw new ConfigException(String.format(
                 "%s config is required by Prometheus metric sampler", PROMETHEUS_SERVER_ENDPOINT_CONFIG));
         }
+        final String basicAuth = (String) configs.get(PROMETHEUS_SERVER_ENDPOINT_BASIC_AUTH_CONFIG);
```

Comment:
Thanks for submitting this PR.

* While basic auth is a good start, I would suggest we introduce an interface called something like ``PrometheusAuthenticator`` with an ``addAuth(httpRequest)`` method. The Sampler can then accept the interface type as an argument instead of taking a String. You can implement a PrometheusBasicAuthenticator and use that implementation. 

* I would also recommend a PrometheusNoOpAuthenticator as a default implementation.
* Add a configuration that can be changed by user to provide the Authenticator class.


Since a lot of AWS MSK customers use the Open Monitoring, they will continue to use the NoOpAuthenticator which would be default. Amazon Managed Prometheus has a sigv4 signing authenticator and if someone is using AMP, then they could write or add a AMP based implementation that signs the http request. See this - https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-query.html

I am not trying to look at it from only AWS perspective but a holistic Prometheus integration with CC. 

More on Prometheus auth can be found here - https://www.robustperception.io/prometheus-security-authentication-authorization-and-encryption/